homeassistant/components/sensor/onewire.py

```diff
@@ -34,63 +50,54 @@
-        for device_family in DEVICE_FAMILIES:
+        for device_family in DEVICE_SENSORS.keys():
             for device_folder in glob(os.path.join(base_dir, device_family +
                                                    '[.-]*')):
-                sensor_ids.append(os.path.split(device_folder)[1])
-                device_files.append(os.path.join(device_folder, 'w1_slave'))
+                sensor_id = os.path.split(device_folder)[1]
+                device_file = os.path.join(device_folder, 'w1_slave')
+                devs.append(OneWire(device_names.get(sensor_id, sensor_id),
+                                    device_file, 'temperature'))
     else:
         for family_file_path in glob(os.path.join(base_dir, '*', 'family')):
             family_file = open(family_file_path, "r")
             family = family_file.read()
-            if family in DEVICE_FAMILIES:
-                sensor_id = os.path.split(
-                    os.path.split(family_file_path)[0])[1]
-                sensor_ids.append(sensor_id)
-                device_files.append(os.path.join(
-                    os.path.split(family_file_path)[0], 'temperature'))
-
-    if device_files == []:
+            if family in DEVICE_SENSORS.keys():
+                for sensor_key, sensor_value in DEVICE_SENSORS[family].items():
+                    sensor_id = os.path.split(
+                        os.path.split(family_file_path)[0])[1]
+                    device_file = os.path.join(
+                        os.path.split(family_file_path)[0], sensor_value)
+                    devs.append(OneWire(device_names.get(sensor_id, sensor_id),
+                                        device_file, sensor_key))
+
+    if devs == []:
         _LOGGER.error("No onewire sensor found. Check if dtoverlay=w1-gpio "
                       "is in your /boot/config.txt. "
                       "Check the mount_dir parameter if it's defined")
         return
 
-    devs = []
-    names = sensor_ids
-
-    for key in config.keys():
-        if key == 'names':
-            # Only one name given
-            if isinstance(config['names'], str):
-                names = [config['names']]
-            # Map names and sensors in given order
-            elif isinstance(config['names'], list):
-                names = config['names']
-            # Map names to ids.
-            elif isinstance(config['names'], dict):
-                names = []
-                for sensor_id in sensor_ids:
-                    names.append(config['names'].get(sensor_id, sensor_id))
-    for device_file, name in zip(device_files, names):
-        devs.append(OneWire(name, device_file))
     add_devices(devs, True)
 
 
 class OneWire(Entity):
     """Implementation of an One wire Sensor."""
 
-    def __init__(self, name, device_file):
+    def __init__(self, name, device_file, stype):
         """Initialize the sensor."""
-        self._name = name
+        self._name = name+' '+stype
```

Comment:
This means that it is no longer possible for people to override the name. We should only append the type if no name was given.

Reply:
The solution in the earlier code led to some issues with assigning the correct values to the right type. Should be ok to just remove this change now.

Reply:
Oh, and removing the change should also lead to removing breaking change status I guess.